### PR TITLE
feat: da compression

### PR DESCRIPTION
**Pull request by hal3e**
thanks @rymnc for the help and the initial PR.

- Importing and storing compressed blocks instead of full fuel blocks
- Removed all code related to full blocks
- Removed block validation and related code
- Refactored tests

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -46,10 +46,10 @@ config = { version = "0.14", default-features = false }
 delegate = { version = "0.13", default-features = false }
 flate2 = { version = "1.0", default-features = false }
 fs_extra = { version = "1.3", default-features = false }
-fuel-core-chain-config = { version = "0.36", default-features = false }
-fuel-core-client = { version = "0.36", default-features = false }
-fuel-core-types = { version = "0.36", default-features = false }
-fuel-crypto = { version = "0.57", default-features = false }
+fuel-core-chain-config = { version = "0.38", default-features = false }
```
Comment (MujkicA) [inline Cargo.toml:49]: ci is using 0.39.0

File: packages/fuel/src/client.rs [hunk 2]
```diff
@@ -1,49 +1,39 @@
...
-        full_blocks_req_size: NonZeroU32,
-    ) -> Self {
+    pub fn new(url: &Url, unhealthy_after_n_errors: usize) -> Self {
         let client = GqlClient::new(url).expect("Url to be well formed");
         Self {
             client,
             metrics: Metrics::default(),
             health_tracker: ConnectionHealthTracker::new(unhealthy_after_n_errors),
-            full_blocks_req_size,
+            num_buffered: NonZeroU32::new(5).expect("is non zero"), //TODO: hal3e make this configurable
```
Comment (MujkicA) [inline packages/fuel/src/client.rs:36]: open todo

File: packages/services/src/block_importer.rs [hunk 3]
```diff
@@ -426,19 +260,16 @@ mod tests {
         Ok(())
     }
 
-    #[tokio::test]
+    #[tokio::test] // TODO: hal3e check this
```
Comment (MujkicA) [inline packages/services/src/block_importer.rs:263]: todo

File: packages/services/src/block_bundler.rs [hunk 4]
```diff
@@ -402,30 +401,35 @@ mod tests {
         Ok(())
     }
 
+    fn merge_data(blocks: impl IntoIterator<Item = CompressedFuelBlock>) -> NonEmpty<u8> {
+        blocks
+            .into_iter()
+            .flat_map(|b| b.data)
```
Comment (MujkicA) [inline packages/services/src/block_bundler.rs:407]: shouldn't we also include the height

File: packages/fuel/src/client/block_ext.rs
Comment (segfault-magnet) [inline packages/fuel/src/client/block_ext.rs:1]: Maybe then remove the `build.rs` as well?

File: packages/services/src/block_bundler.rs (hunk 4 above)
Comment (segfault-magnet) [reply packages/services/src/block_bundler.rs:407]: it's probably inside of the compressed block itself, should be checked @hal3e

Comment (hal3e) [reply packages/services/src/block_bundler.rs:407]: Compressed block contains `PartialBlockHeader` which has the height. https://docs.rs/fuel-core-types/latest/fuel_core_types/blockchain/header/struct.PartialBlockHeader.html

**Review by segfault-magnet — APPROVED** [review]

**Review by digorithm — APPROVED** [review]

**Review by MujkicA — APPROVED** [review]

**Pull request merged**